lib/Driver/Job.cpp

```diff
@@ -22,28 +22,82 @@
 using namespace swift;
 using namespace swift::driver;
 
+StringRef CommandOutput::getOutputForInputAndType(StringRef PrimaryInputFile,
+                                                  types::ID Type) const {
+  auto const *M = DerivedOutputMap.getOutputMapForInput(PrimaryInputFile);
+  if (!M)
+    return StringRef();
+  auto const Out = M->find(Type);
+  if (Out == M->end())
+    return StringRef();
+  return StringRef(Out->second);
+}
+
+void CommandOutput::addEntryOrCheckConflict(StringRef PrimaryInputFile,
+                                            types::ID type,
+                                            StringRef OutputFile) {
+  auto &M = DerivedOutputMap.getOrCreateOutputMapForInput(PrimaryInputFile);
+  auto const Out = M.find(type);
+  if (Out == M.end()) {
+    M.insert(std::make_pair(type, OutputFile));
```

Comment:
You can use `insert`'s return value to save a hash lookup here. It's probably not that important in this case, but still, a habit to get into.

Reply:
Done.